src/test/java/org/tron/p2p/utils/NetUtilTest.java

```diff
@@ -75,6 +78,22 @@ public void testExternalIp() {
     Assert.assertFalse(ip.startsWith("172.31."));
   }
 
+  @Test
+  public void testGetIP() {
+    long t1 = System.currentTimeMillis();
+    try {
+      Method method = NetUtil.class.getDeclaredMethod("getExternalIp", String.class);
+      method.setAccessible(true);
+      method.invoke(NetUtil.class, Constant.ipV4Urls.get(0));
+    } catch (Exception e) {
+      Assert.fail();
+    }
+    long t2 = System.currentTimeMillis();
+    NetUtil.getExternalIpV4();
+    long t3 = System.currentTimeMillis();
+    Assert.assertTrue(t2 - t1 >= t3 - t2);
```

Comment:
Results are randomized.

Reply:
What do you mean "randomized"?

Reply:
`t2 - t1 >= t3 - t2` doesn't always hold true.

Reply:
I have tried this many times, it works well though network is not avaiIable. But i comment this. Add test case of compare ip from different source.

Reply:
@chengtx01 @halibobo1205  If the node has multi outer extranet ip, querying from different urls may get different ip. Is there any problem?